### DISCUSSION TITLE
Problem with permanentRedirect

### DISCUSSION THREAD

**Discussion by meridPL** [vercel/next.js#66311] [Help]
### Summary

Hi,
I use an app router and on each pages, I add 
`export const dynamic = 'force-static';`
I test even without this line and also I have that same case.
The URL page I paste to https://httpstatus.io/ should be redirected to a new page with status 308 -> 200.
Unfortunately, on the first time I get 308 and then 200, but each next time I get only 308. Why?
From API I get information about redirect and I use     
`permanentRedirect(redirectUrl, RedirectType['push']);`

**Comment by meridPL** [comment]
Probably, I found solution. I moved logic from server page to middlware and added
```
return NextResponse.redirect(new URL(url, req.url), {
      status: 308,
    });
```

**Reply by serdar-webdev** [reply]
Hey, if you want to redirect, simple way is add it to next config js;
if there will be more than 10k requests so it will be right to use middleware to redirect;

**Comment by meridPL** [comment]
Yes, I know but those information get from API when I have to redirect page and I don't make it in next.config.js
